src/utils/duplicateDates.js

```diff
@@ -0,0 +1,20 @@
+
+function duplicateDates(dates, date) {
```

Comment:
the name of the function mismatches with the last for each logic, its not only for duplicated dates.